config/confighttp/confighttp.go

```diff
@@ -82,11 +82,16 @@ type HTTPClientSettings struct {
 	IdleConnTimeout *time.Duration `mapstructure:"idle_conn_timeout"`
 }
 
-// DefaultHTTPClientSettings returns HTTPClientSettings type object with
+// Deprecated: [v0.45.0] Use NewDefaultHTTPClientSettings instead.
+func DefaultHTTPClientSettings() HTTPClientSettings {
+	return NewDefaultHTTPClientSettings()
```

Comment:
Using alias will make codecov happy
```
var DefaultHTTPClientSettings = NewDefaultHTTPClientSettings
```

Reply:
If this works, mind if I update the recommendation inside the contribution guide?

Reply:
Go for it in a separate PR :)